mono/utils/mono-error-internals.h

```diff
@@ -284,4 +284,49 @@ mono_error_set_specific (MonoError *error, int error_code, const char *missing_m
 void
 mono_error_set_first_argument (MonoError *oerror, const char *first_argument);
 
+// Provide single instruction inlinable forms of GetLastError and SetLastError.
+// Nobody should worry about their cost then.
+// Offsets are from Wikipedia.
+// Runtime disassembly could assert correctness.
+
+// FIXME include this file more so the replacement takes affect.
+
+#if HOST_WIN32 && (HOST_X86 || HOST_AMD64)
+
+#include <windows.h>
+
+// Naming violation so can search disassembly for GetLastError / SetLastError.
+#define GetLastError mono_GetLastError
+#define SetLastError mono_SetLastError
+
+static inline
+unsigned long
+__stdcall
+GetLastError (void)
+{
+#if HOST_X86
+    return __readfsdword (0x34);
+#elif HOST_AMD64
+    return __readgsdword (0x68);
+#else
+#error unknown architecture
```

Comment:
Can we just rely on the system headers for unknown architecture instead of generating errors? (Windows/ARM64)

Reply:
Oh, now I see that is actually done. It is a bit misleading :)

Reply:
It was probably accurate earlier. I changed the message.
Perhaps could use more commenting/clarity.